### PR TITLE
Remove diesel OID priming

### PR DESCRIPTION
**Pull request by jgallagher**
See #5561 for rationale. We could add this back if we could figure out how to correctly invalidate the cache after a schema upgrade, but there don't seem to be any easy ways to do that. This will ensure correctness for now and give us time to figure out a different path.

### CODE REVIEW

File: nexus/db-model/src/schema_versions.rs [hunk 1]
```diff
@@ -12,6 +12,9 @@ use omicron_common::api::external::SemverVersion;
 use once_cell::sync::Lazy;
 use std::collections::BTreeMap;
 
+pub const SCHEMA_DIR: &str =
```
Comment (davepacheco) [inline nexus/db-model/src/schema_versions.rs:15]: Why did you need to move this?

**Review by davepacheco — APPROVED** [review]

File: nexus/db-model/src/schema_versions.rs (hunk 1 above)
Comment (davepacheco) [reply nexus/db-model/src/schema_versions.rs:15]: Nevermind -- found it!

File: nexus/db-queries/src/db/pool_connection.rs [hunk 2]
```diff
@@ -224,90 +39,152 @@ impl CustomizeConnection<Connection<DbConnection>, ConnectionError>
...
+        {
+            let client = db.connect().await.expect("Failed to connect to CRDB");
+            client
+                .batch_execute(DISALLOW_FULL_TABLE_SCAN_SQL)
+                .await
+                .expect("failed to disallow full table scans");
+
+            let all_versions = AllSchemaVersions::load(SCHEMA_DIR.into())
+                .expect("Failed to load schema versions");
+            for version in all_versions.iter_versions() {
```
Comment (davepacheco) [inline nexus/db-queries/src/db/pool_connection.rs:101]: It does make me a little nervous to duplicate this logic.  I wonder if we should modify the existing tests that do this instead?

File: nexus/db-queries/src/db/pool_connection.rs [hunk 3]
```diff
@@ -224,90 +39,152 @@ impl CustomizeConnection<Connection<DbConnection>, ConnectionError>
...
+        // Attempt to insert a sled resource. This involves the
+        // `sled_resource_kind` enum, whose OID was changed by the schema
+        // migration in version 53.0.0. If the diesel OID cache was populated
+        // when we the datastore started polling for the version, this would
+        // fail with a `type with ID $NUM does not exist` error.
+        //
+        // This test isn't perfect: (a) it's possible on an overloaded system
+        // the datastore doesn't start polling CRDB until after we've finished
+        // applying updates (although that's slow, so this seems very unlikely),
+        // and (b) it's possible some _other_ OID has been cached that we're not
```
Comment (davepacheco) [inline nexus/db-queries/src/db/pool_connection.rs:166]: (a) seems like a real problem that will lead to a flaky test.

I'm not sure how much plumbing you'd need to do, but what if you create the pool first here, check out a connection (so you know you've got one), then use that pool to create a new datastore?  Then you don't need a background task and there's no race.

**Review by davepacheco — CHANGES_REQUESTED** [review]

File: nexus/db-queries/src/db/pool_connection.rs (hunk 2 above)
Comment (davepacheco) [reply nexus/db-queries/src/db/pool_connection.rs:101]: (This one's not a blocker for me.)

Comment (jgallagher) [reply nexus/db-queries/src/db/pool_connection.rs:101]: The existing tests that do this don't use a connection pool. I guess they could? I'll look and see (taking the other comment into consideration too while I'm there).

**Comment by jgallagher** [comment]
With apologies for the force push (it was easier to start back on `main` to ensure the new test failed while we still had caching in place), both your comments should be addressed now. I moved the test to the existing schema migration test, which muddies it up a bit but is probably better than duplicating the schema migration logic, and it builds a `Pool` by hand to avoid the datastore raciness.

**Review by davepacheco — APPROVED** [review]
